### PR TITLE
WIKI-1248: HOW-TO Guide template and Three-Column Layout template doesn't work

### PR DESCRIPTION
**Pull request by azaoui**
WIKI-1248 HOW-TO Guide template and Three-Column Layout template doesn't work

### CODE REVIEW

**Comment by azaoui** [comment]
description : skip macro content from encoding as a trusted third party

File: commons-component-common/src/main/java/org/exoplatform/commons/utils/StringCommonUtils.java [hunk 1]
```diff
@@ -81,7 +85,51 @@ public static String encodeScriptMarkup(String input) {
...
+        startIndex.add(macroMatcher.start());
+        endIndex.add(macroMatcher.end());
+      }
+      boolean skip = false;
+      while (matcher.find()) {
+        int styleStartPos = matcher.start();
+        for (int i = 0; i < startIndex.size(); i++) {
+          int firstIndex = startIndex.get(i);
+          int lastIndex = endIndex.get(i);
+          if (styleStartPos > firstIndex && styleStartPos < lastIndex)
```
Comment (ahamdi) [inline commons-component-common/src/main/java/org/exoplatform/commons/utils/StringCommonUtils.java]: Please use brackets and remove the two empty lines

File: commons-component-common/src/test/java/org/exoplatform/commons/utils/StringCommonUtilsTest.java [hunk 2]
```diff
@@ -17,6 +17,7 @@
 package org.exoplatform.commons.utils;
 
 import junit.framework.TestCase;
+
```
Comment (ahamdi) [inline commons-component-common/src/test/java/org/exoplatform/commons/utils/StringCommonUtilsTest.java]: remove this empty line

File: commons-component-common/src/test/java/org/exoplatform/commons/utils/StringCommonUtilsTest.java [hunk 3]
```diff
@@ -69,6 +70,67 @@ public void testEncodeScriptMarkup() {
     assertEquals("<BASE HREF=\"('XSS');\">", StringCommonUtils.encodeScriptMarkup(input11));
   }
   
+  
+  public void testencodeWikiScriptMarkup() {
```
Comment (ahamdi) [inline commons-component-common/src/test/java/org/exoplatform/commons/utils/StringCommonUtilsTest.java:73]: your test should include both malicious code and macro code to make sure that the sanitization is working as axpected

**Comment by azaoui** [comment]
done

**Pull request closed without merging**